docs-src/ref-programs.scrbl

```diff
@@ -2297,3 +2297,33 @@ it does not matter who @reachin{publish}es, such as in a @reachin{timeout}.
 @reachin{Anybody} is strictly an abbreviation of a @reachin{race} involving all of the named participants of the application.
 In an application with a @tech{participant class}, this means any principal at all, because there is no restriction on which principals (i.e. addresses) may serve as a member of that class.
 In an application without any @tech{participant class}es, @reachin{Anybody} instead would mean only the actual previously-bound @tech{participant}s.
+
+@subsubsection{'use strict'}
+
+@(mint-define! '("'use strict'"))
+@reach{
+  'use strict'; }
+
+@index{'use strict'} @reachin{'use strict'} enables unused variables checks for all subsequent
+declarations within the current scope. If a variable is declared, but never used, there will
+be a warning emitted at compile time.
+
+Strict mode will reject some code that is normally valid and limit how dynamic Reach's type system is.
+For example, normally Reach will permit expressions like the following to be evaluated:
+
+@reach{
+  const foo = (o) =>
+    o ? o.b : false;
+
+  void foo({ b: true });
+  void foo(false); }
```

Comment:
"Reach allows `o` to be either an object with a `b` field or `false, because it partially evaluates the program at compile time."